### PR TITLE
Improve parsing of non-nn.Sequential PyTorch models

### PR DESCRIPTION
**Pull request by vloncar**
# Description

In case of skipped layers, like Flatten or Dropout, PyTorch converter will incorrectly parse the model inputs, we need to create an input map similar to how Keras handles it. This was the case in #839. Additionally, as observed in #838, parsing of BN weights was broken. These fixes are cherrypicked from my development branch for parsing GNNs, not fully tested standalone, so I'm making this a draft PR for now before I add proper tests.

## Type of change

- [x] Bug fix (non-breaking change that fixes an issue)

## Tests

Currently lacking. Will add something along the lines of code shared in #838 and #839

## Checklist

- [x] I have read the [guidelines for contributing](https://github.com/fastmachinelearning/hls4ml/blob/main/CONTRIBUTING.md).
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] I have installed and run `pre-commit` on the files I edited or added.
- [ ] I have added tests that prove my fix is effective or that my feature works. <-- Will do in a follow-up commit

### CODE REVIEW

**Comment by jmitrevs** [comment]
Note, Flatten is not a skip layer for Keras, but gets turned into a reshape. We should check why we made that decision and a different decision here.

**Comment by jmitrevs** [comment]
Concerning Flatten always disappearing, whether io_stream or io_parallel, I think effectively the optimizers do that now, but I think we were worried that this isn't guaranteed to always be the case. More qualifications can be added in the streaming case if this isn't true, or if we have a different backend. I am not sure what's best but in general I think we should handle Keras and Pytorch the same way, unless there's a good reason for this to not be the case.

**Comment by vloncar** [comment]
Continued in #848. Closing.
